### DISCUSSION TITLE
getCameraOrbit not a function error

### DISCUSSION THREAD

**Discussion by SteveTou** [google/model-viewer#4511] [Q&A] [answered]
Quick background- I am trying to use buttons to orbit around or orient the model, because I've found it to be clunky using swipes on mobile devices as sometimes the swipe moves the model, other times the swipe moves the page, other times it moves the browser (Safari on iPhone).

1. I cannot figure out a way to access the orientation property in order change it with button click(s).
2. I have tried using the getCameraOrbit method, but I am getting an error in Chrome Dev that says getCameraOrbit is not a function. 

(I understand camera orbit and orientation are two different things- I'm trying both to test best results).
Using xampp, Chrome.  Any thoughts on how I can best tackle my issues and achieve the goal?

Direct copy and paste (notepad++) of relevant code for #2-
 const orbit = mv.getCameraOrbit();
![image](https://github.com/google/model-viewer/assets/67981473/cb910b80-f57b-4874-981f-8f3e08ab3fa6)

**Comment by elalish** [comment]
Looks like the only problem is you haven't found the MV element. Check your query. I find it's quickest to debug by running a few lines right in the JS console. If `mv.src` doesn't give you your GLB path, you know you haven't found the right `mv` yet.

**Reply by SteveTou** [reply]
Hmmm... I don't think that's it.  I get the element for other things which work-->
JS
      const mv = document.querySelector("model-viewer#toggle-model"); <--1st line in script.
     document.querySelector("#yaw15").addEventListener('click', () => {mv.cameraOrbit = 'calc(75deg + 15deg) 75deg 100%';}); <--Works!
    const orbit = mv.getCameraOrbit(); <--Throws error!

HTML
     <model-viewer id="toggle-model" ...>
     `<button` id="yaw15"> `Y15</button>`

Any ideas?  Thanks for the quick reply.

**Reply by elalish** [reply]
Your script is probably running before the model-viewer.js has loaded. You may want to look into the web standards for custom elements and script execution order. You can verify this by checking that your script runs properly in the console (once everything is loaded).

**Reply by SteveTou** [reply]
Ok, so I type directly in the console and get-
![image](https://github.com/google/model-viewer/assets/67981473/2918ad5d-0505-410f-a997-4ffb6ba263e3)

I guess it must be some loading order issue, but I don't know how to fix it.  Any ideas?

**Reply by SteveTou** [reply, marked as answer]
Thanks for the direction- figured it out.  I have my scripts at the bottom of my file, (inside the closing body tag), but I did not have type="module" attribute, so as you said, they were firing before the model-viewer.js, which I have farther up in the file, and also has the type="module" attribute.  Thanks for the help!!
